### ISSUE TITLE
TypeScript support

### ISSUE THREAD

**Issue by Traxo7** [FiguredLimited/vue-mc#48]
Do you plan to add TypeScript definition file?

**Comment by rtheunissen** [comment]
We currently don't have any plans to add this, to my knowledge.

**Comment by rtheunissen** [comment]
I would love to go full TS as soon as possible. 👍

**Comment by BehindTheMath** [comment]
I would like to help with this. I've actually already written definition files, and I was going to submit them to DefinitelyTyped, but your idea is better.

I just don't know anything about Rollup, and if it works with TS. I suppose I could research that.

**Comment by BehindTheMath** [comment]
Do you prefer explicit typing everywhere, or to use as much type inference as possible?

**Comment by BehindTheMath** [comment]
@rtheunissen I'm basically finished converting the source files. I'm waiting for #84, #86, and #87 before making a PR.

I also have a few more questions:
1. Do you want Model and Collection to be abstract, to force them to be subclassed?
2. Do you want the tests to be written in TS as well, or leave them in JS and test the transpiled source files?

**Comment by rtheunissen** [comment]
> Do you want Model and Collection to be abstract, to force them to be subclassed

I think a generic model and collection should exist, so to me I think no.

> Do you want the tests to be written in TS as well, or leave them in JS and test the transpiled source files?

I think we should test as plain JS, so that we can be sure that the behaviour is tested in a non-TS environmnent, and assume that it would be equivalent with TS. We can always convert the tests at a later time if there is a good reason to do so.

**Comment by rtheunissen** [comment]
@BehindTheMath all three of those now resolved. 👍

**Comment by rtheunissen** [comment]
@BehindTheMath what is the status around TS support?

**Comment by BehindTheMath** [comment]
I have to look over my code. I'm going to have to rebase on v0.5.0, but I think the actual TS code is basically done; it's the tooling that needs to be polished. You can see the code [here](https://github.com/BehindTheMath/vue-mc/commits/feature/typescript).

What happened with #85?

**Comment by Flame1994** [comment]
Any update on this?

**Comment by JonLev** [comment]
It seems that version v0.6.0 is out ! @BehindTheMath do you have any other requirements?

**Comment by BehindTheMath** [comment]
I'm working on rebasing on v6.0.0.

Since there are no existing type definitions for `lodash/stubTrue`, I submitted DefinitelyTyped/DefinitelyTyped#38920 to add them.

Once the rebasing is finished, I can take another look at the tooling, which I never finished properly.

**Comment by NViktors** [comment]
Any update on this? Can't wait for TypeScript support 👀

**Comment by BehindTheMath** [comment]
I'm waiting for a resolution to #132. 

/cc @rtheunissen

**Comment by BehindTheMath** [comment]
My PR is finished, and is ready to be reviewed and merged.

**Comment by JonLev** [comment]
you're my hero @BehindTheMath !
@rtheunissen can this be in the next release ?

**Comment by ilawton** [comment]
Any news?

**Comment by ilawton** [comment]
Any news?

**Comment by rtheunissen** [comment]
> you're my hero @BehindTheMath !
> @rtheunissen can this be in the next release ?

Absolutely, we are working on it. Thank you @BehindTheMath

**Comment by albertpratomo** [comment]
> > you're my hero @BehindTheMath !
> > @rtheunissen can this be in the next release ?
> 
> Absolutely, we are working on it. Thank you @BehindTheMath

Any update about this? Would love to use the TS version 😃

**Comment by Acen** [comment]
It's now been merged into master. 
@rtheunissen will just need to tag a release. Are there breaking changes with the intro of TS? 

Does the install/running config change at all?
